### PR TITLE
Feat: create new aggratedMeasureData Process from command

### PR DESCRIPTION
**Pull request by MadsDue**
## References

* #0000

### CODE REVIEW

File: source/ApplyDBMigrationsApp/ApplyDBMigrationsApp.csproj [hunk 1]
```diff
@@ -99,6 +99,9 @@ limitations under the License.
     <EmbeddedResource Include="Scripts\Model\202306090924 Change archived id type to nvarchar.sql" />
     <EmbeddedResource Include="Scripts\Model\202305161024 Add Process Type to ArchivedMassage.sql" />
     <EmbeddedResource Include="Scripts\Model\202305171026 Rename Process Type to BusinessReason.sql" />
+    <EmbeddedResource Include="Scripts\Model\202306271334 Create AggregatedMeasureDataProcesses Table.sql">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (lasrinnil) [inline source/ApplyDBMigrationsApp/ApplyDBMigrationsApp.csproj:103]: Think its safe to remove line 103-104

File: source/Domain/Transactions/ProcessId.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Domain.SeedWork;
+
+namespace Domain.Transactions;
+
+public class ProcessId : ValueObject
```
Comment (lasrinnil) [inline source/Domain/Transactions/ProcessId.cs:19]: This class appears to contain the same logic as `TransactionId`, should we consider a rename of that instead?

File: source/ApplyDBMigrationsApp/Scripts/Model/202306271334 Create AggregatedMeasureDataProcesses Table.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿BEGIN
+    CREATE TABLE [dbo].[AggregatedMeasureDataProcesses]
+    (
+        [RecordId]                                      [int] IDENTITY (1,1) NOT NULL,
+        [ProcessId]                                     [nvarchar](50)       NOT NULL,
+        [SettlementSeriesVersion]                       [nvarchar](50)       NULL,
```
Comment (lasrinnil) [inline source/ApplyDBMigrationsApp/Scripts/Model/202306271334 Create AggregatedMeasureDataProcesses Table.sql:6]: All the values here which allow `NULL`, would they still represent a valid domain model? Does the schema even allows properties such as `StartDateAndOrTimeDateTime`, `EndDateAndOrTimeDateTime` to not have a value?

File: source/Domain/Transactions/AggregatedMeasureData/IAggregatedMeasureDataProcessRepository.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Domain.Transactions.AggregatedMeasureData
+{
+    /// <summary>
+    /// Storage for transactions
+    /// </summary>
+    public interface IAggregatedMeasureDataProcessRepository
+    {
+        /// <summary>
+        /// Adds a transaction to store
```
Comment (lasrinnil) [inline source/Domain/Transactions/AggregatedMeasureData/IAggregatedMeasureDataProcessRepository.cs:26]:
```suggestion
        /// Adds a new process to store
```

File: source/Domain/Transactions/AggregatedMeasureData/IAggregatedMeasureDataProcessRepository.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Domain.Transactions.MoveIn;
+using NodaTime;
+
+namespace Domain.Transactions.AggregatedMeasureData
+{
+    /// <summary>
+    /// Storage for transactions
```
Comment (lasrinnil) [inline source/Domain/Transactions/AggregatedMeasureData/IAggregatedMeasureDataProcessRepository.cs:21]:
```suggestion
    /// Storage for Process
```

File: source/Infrastructure/Configuration/CompositionRoot.cs [hunk 6]
```diff
@@ -81,7 +80,6 @@ private CompositionRoot(IServiceCollection services)
             services.AddScoped<ITransactionIds, TransactionIdRegistry>();
             services.AddScoped<IMessageIds, MessageIdRegistry>();
             services.AddScoped(typeof(IMessageQueueDispatcher<>), typeof(MessageQueueDispatcher<>));
-            services.AddScoped<IMoveInTransactionRepository, MoveInTransactionRepository>();
```
Comment (lasrinnil) [inline source/Infrastructure/Configuration/CompositionRoot.cs:84]: I'm reading this as if this line has been removed, is that correct?

File: source/Infrastructure/Transactions/MoveIn/MoveInConfiguration.cs [hunk 7]
```diff
@@ -64,6 +65,7 @@ internal static class MoveInConfiguration
         }
 
         services.AddScoped<MoveInNotifications>();
+        services.AddScoped<IMoveInTransactionRepository, MoveInTransactionRepository>();
```
Comment (lasrinnil) [inline source/Infrastructure/Transactions/MoveIn/MoveInConfiguration.cs:68]: Found it :)

File: source/ApplyDBMigrationsApp/Scripts/Model/202306271334 Create AggregatedMeasureDataProcesses Table.sql (hunk 3 above)
Comment (lasrinnil) [reply source/ApplyDBMigrationsApp/Scripts/Model/202306271334 Create AggregatedMeasureDataProcesses Table.sql:6]: The `Serie` from which the Process is created from does not have any nullable values

File: source/Domain/Transactions/AggregatedMeasureData/AggregatedMeasureDataProcess.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    public class AggregatedMeasureDataProcess : Entity
+    {
+#pragma warning disable CS0414
+        private readonly State _state;
+#pragma warning restore CS0414
+
+        private readonly ActorNumber _requestedByActorId;
+
+        public AggregatedMeasureDataProcess(
+            ProcessId processId,
```
Comment (lasrinnil) [inline source/Domain/Transactions/AggregatedMeasureData/AggregatedMeasureDataProcess.cs:30]: When we receive a RSM-016 we should be delivered a `OriginalTransactionId` which we are required to respond on when creating a RSM-014 respons

Comment (MadsDue) [reply source/Domain/Transactions/AggregatedMeasureData/AggregatedMeasureDataProcess.cs:30]: This is represented by the businessTransactionId

**Review by lasrinnil — APPROVED** [review]

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**